extensions/wikia/AdEngine/js/context/adContext.js

```diff
@@ -170,6 +184,22 @@ define('ext.wikia.adEngine.adContext', [
 		}
 	}
 
+	function shouldLoadTaboolaOnBlockingTraffic(taboolaConfig) {
+		var i = 0,
+			taboolaSlot;
+
+		for (taboolaSlot in taboolaConfig) {
+			if (taboolaConfig.hasOwnProperty(taboolaSlot) && taboolaConfig[taboolaSlot].recovery) {
+				for (i=0; i<taboolaConfig[taboolaSlot].recovery.length; i++) {
+					if (geo.isProperGeo(taboolaConfig[taboolaSlot].recovery[i])) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ANestedIfDepth)